src/hooks/safe-apps/useSafeAppFromManifest.ts

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect } from 'react'
+import { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import { Errors, logError } from '@/services/exceptions'
+import { fetchSafeAppFromManifest } from '@/services/safe-apps/manifest'
+
+type UseSafeAppFromManifestReturnType = {
+  safeApp?: SafeAppData
+  isLoading: boolean
+}
+
+const useSafeAppFromManifest = (appUrl: string, chainId: string): UseSafeAppFromManifestReturnType => {
+  const [safeApp, setSafeApp] = useState<SafeAppData>()
+  const [isLoading, setIsLoading] = useState(false)
+
+  useEffect(() => {
+    if (!appUrl) {
+      return
+    }
+
+    const loadApp = async () => {
+      setIsLoading(true)
+      try {
+        const app = await fetchSafeAppFromManifest(appUrl, chainId)
+        setSafeApp(app)
+      } catch (e) {
+        logError(Errors._900, `${appUrl}, ${(e as Error).message}`)
+      } finally {
+        setIsLoading(false)
+      }
+    }
+
+    loadApp()
+  }, [appUrl, chainId])
```

Comment:
You can use `useAsync` for this.

Reply:
It's already used in the form that adds a custom safe app.

Reply:
ok i'll take a look 🤗

Reply:
beautiful! done